rkt/api_service.go

```diff
@@ -333,6 +332,43 @@ func (s *v1AlphaAPIServer) getPodManifest(p *pod) (*schema.PodManifest, []byte,
 	return &manifest, data, nil
 }
 
+// getPodCgroup gets the cgroup path for a pod. This can be done in one of a couple ways:
+//
+// 1) stage1-coreos/stage1-kvm
+//
+// This one can be tricky because after machined registration, the cgroup might change.
+// For both of these flavors, the cgroup it will end up in after machined moves
+// it is written to a file named `subcgroup`, so we use that instead.
+//
+// 2) All others
+//
+// Assume that we cna simply get the cgroup of the pod's init PID, and use that.
+func getPodCgroup(p *pod, pid int) (string, error) {
+	// Note, this file should always exists if the pid is known since it is
+	// written before the pid file
+	data, err := p.readFile("subcgroup")
+	if err == nil {
+		return "/" + string(data), nil
+	}
+
+	if !os.IsNotExist(err) {
+		return "", err
+	}
+	// if it is an "isNotExist", assume this is a stage1 flavor that won't change
+	// cgroups. Just give the systemd cgroup of its pid
+	cgroup, err := cgroup.GetCgroupPathByPid(pid, "name=systemd")
```

Comment:
@euank OK, can we add a comment to mention this?
Seems this return error on non-systemd distros, cc @alban @lucab @iaguis